src/Utilities/Compiler/WellKnownTypeProvider.cs

```diff
@@ -126,5 +191,71 @@ internal bool IsTaskOfType([NotNullWhen(returnValue: true)] ITypeSymbol? typeSym
                 && namedTypeSymbol.TypeArguments.Length == 1
                 && typeArgumentPredicate(namedTypeSymbol.TypeArguments[0]);
         }
+
+        private static string[] GetNamespaceNamesFromFullTypeName(string fullTypeName)
+        {
+            int dotCount = 0;
+            int lastDotIndex = -1;
+            for (int i = 0; i < fullTypeName.Length; i++)
+            {
+                switch (fullTypeName[i])
+                {
+                    case '.':
+                        lastDotIndex = i;
+                        dotCount++;
+                        break;
+
+                    // Dunno if Compilation.GetTypeByMetadataName() or IAssemblySymbol.GetTypeByMetadataName() will ever 
+                    // support something like "System.Collections.Generic.List<System.String>", but if they do, we'll be ready.
+                    case '`':
```

Comment:
This list makes the code seem fragile. Should we consider the reverse where we process all known characters, such as valid identifier character and `.`, and bail out on everything else? Probably something like this for the former: http://sourceroslyn.io/#Microsoft.CodeAnalysis/InternalUtilities/UnicodeCharacterUtilities.cs,51

Reply:
Sure, I'll do that. I was copying http://sourceroslyn.io/#Microsoft.CodeAnalysis/MetadataReader/MetadataHelpers.cs,77 before, but I agree known characters will be more robust in the future.